feat: Introduce `managed` flag to KymaCR `.spec.modules` (#1652)
kyma-project/lifecycle-manager: docs/technical-reference/api/kyma-cr.md

      - name: kyma-project.io/module/sample
    ```

### **.spec.modules[].managed**

For each module, it must be defined whether it is `managed` or `unmanaged`.

> If it is `managed`, the Lifecycle Manager will take care of the lifecycle (creating, updating, deleting) the module.

Reviewer: ```suggestion
If it is `managed`, Lifecycle Manager takes care of the module's lifecycle (creating, updating, deleting).
```

> If it is `unmanaged`, the Lifecycle Manager will ignore the module and the users must take care of the lifecycle themselves.

Reviewer: ```suggestion
If it is `unmanaged`, Lifecycle Manager ignores the module and the users must take care of the lifecycle themselves.
```

> To define this, the **managed** flag must be set to either `true` or `false`.

Reviewer: ```suggestion
To define this, the **managed** flag must be either `true` or `false`.
```

By default, the **managed** flag is set to `true` and the module is therefore `managed` by Lifecycle Manager.

```yaml
spec:
  channel: regular
  modules:
  - name: module-name-from-label
    managed: true
```

### **.spec.modules[].customResourcePolicy**

A module may be initialized with preconfigured default data.

> To control this, the **customResourcePolicy** flag is used.

Reviewer: ```suggestion
To control the configuration, use the **customResourcePolicy** flag.
```

> It can either be `CreateAndDelete` or `Ignore`.

Reviewer: ```suggestion
The value can either be `CreateAndDelete` or `Ignore`.
```

While `CreateAndDelete` causes the ModuleTemplate CR's **.spec.data** to be created and deleted to initialize a module with preconfigured defaults, `Ignore` can be used to only initialize the operator without initializing any default data.
This allows users to be fully flexible in regard to when and how to initialize their module.

> By default, the **customResourcePolicy** flag is `CreateAndDelete` and the module is therefore preconfigured.

Reviewer: ```suggestion
By default, the **customResourcePolicy** flag is `CreateAndDelete` which makes the module preconfigured.
```
